articles/css-flexbox-vs-css-grid/index.md

```diff
@@ -0,0 +1,767 @@
+
+### CSS Grid vs CSS Flexbox
+
+Let's have a look at some of the common differences between the two layouts.
+
+#### One vs two dimensions
+
+Flexbox arranges items in a one-dimensional layout. This can be either in rows or columns.
+
+![](/engineering-education/css-flexbox-vs-css-grid/flex-horizontal-or-vertical.jpg)
+[***Image Source***](https://www.youtube.com/watch?v=MdPYenT89o4)
+
+A Flexbox layout fits best when aligning items either horizontally or vertically. It is one dimensional and can only fit a row or a column.
+
+![](/engineering-education/css-flexbox-vs-css-grid/vertical.png)
+[***Image Source***](https://www.youtube.com/watch?v=MdPYenT89o4)
+
+![](/engineering-education/css-flexbox-vs-css-grid/horizontal.png)
+[***Image Source***](https://www.youtube.com/watch?v=MdPYenT89o4)
+
+A perfect example of a one-dimensional layout is a header navigation bar or a footer.
+
+Let's have a look at implementing a header navigation menu using Flexbox.
+
+```html
+<div class="nav">
+    <div>HOME</div>
+    <div>ABOUT</div>
+    <div>CONTACT</div>
+    <div>SIGN UP</div>
+    <div>SIGN IN</div>
+</div>
+```
+
+Some CSS to style the header:
+
+```css
+.nav {
+    display: flex;
+    background: seagreen;
+    height: 50px;
+    align-items: center;
+    gap: 10px;
+}
+
+.nav div {
+    color: whitesmoke;
+    margin-left: 20px;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/flex-header-nav.png)
+
+Most common header layouts lay their items in one direction; thus, it's much easier to use Flexbox because it arranges items in one direction.
+
+On the other hand, a Grid is a two-dimensional layout, meaning you can lay out items in rows and columns.
+
+![](/engineering-education/css-flexbox-vs-css-grid/grid.png)
+[***Image Source***](https://www.youtube.com/watch?v=MdPYenT89o4)
+
+An example of a basic two-dimensional layout.
+
+```html
+<div class="container">
+    <div class="item item1"></div>
+    <div class="item item2"></div>
+    <div class="item item3"></div>
+    <div class="item item4"></div>
+    <div class="item item5"></div>
+    <div class="item item6"></div>
+</div>
+```
+
+```css
+.container {
+    display: grid;
+    grid-template-columns: 150px 150px 150px;
+    grid-gap: 1rem;
+    grid-template-rows: 150px 150px;
+}
+
+.item {
+    background-color: #1EAAFC;
+    color: #fff;
+    border: 6px solid #171717;
+}
+```
+
+![](/engineering-education/css-flexbox-vs-css-grid/grid-example.jpg)
+
+We specify a grid with `display: grid` and then determine rows and columns with `grid-template-columns` and `grid-template-rows`. This defines a two-dimensional layout. In this case, each column and row take the size of `150px`.
+
+#### Content vs layout
+
+Flexbox is content-based; content comes first. By content first, we mean it helps you align the contents of and move blocks and items within a container.
+
```

Comment:
I think I had already fixed this @louisefindlay23.  `content-based; content comes`  i didn't include `Flexbox is content-based so content comes first` as you suggested. This was highlighted by Grammarly, and I update it as suggested by Grammarly. Moreover  `Flexbox is content-based so content comes first` need `content`  to be denoted with an article `the` as  `Flexbox is content-based so the content comes first`. And this is how Grammarly suggests for that sentence `Flexbox is content-based, so the content comes first.`

But either way, I've updated that section as you suggested.